stdlib/public/core/IntegerParsing.swift.gyb

```diff
@@ -67,21 +67,20 @@ internal func _parseUnsignedAsciiAsUIntMax(
 /// non-negative number <= `maximum`, return that number.  Otherwise,
 /// return `nil`.
 ///
-/// - Note: If `text` begins with `"+"` or `"-"`, even if the rest of
-///   the characters are `"0"`, the result is `nil`.
+/// - Note: For text matching the regular expression "-0+", the result
+///   is `0`, not `nil`.
 internal func _parseAsciiAsUIntMax(
-  u16: String.UTF16View, _ radix: Int, _ maximum: UIntMax
+  utf16: String.UTF16View, _ radix: Int, _ maximum: UIntMax
 ) -> UIntMax? {
-  if u16.isEmpty { return nil }
-  let c = u16.first
-  if _fastPath(c != _ascii16("-")) {
-    let unsignedText
-      = c == _ascii16("+") ? u16.dropFirst() : u16
-    return _parseUnsignedAsciiAsUIntMax(unsignedText, radix, maximum)
-  }
-  else {
-    return _parseAsciiAsIntMax(u16, radix, 0) == 0 ? 0 : nil
-  }
+  if utf16.isEmpty { return nil }
+  // Parse (optional) sign
+  let (digitsUTF16, hasMinus) = _parseOptionalAsciiSign(utf16)
+  // Parse digits
+  guard let result = _parseUnsignedAsciiAsUIntMax(digitsUTF16, radix, maximum) else { return nil }
+  // Disallow < 0
+  if hasMinus && result != 0 { return nil }
+  // Return
```

Comment:
The comments are intended as headings for code grouping (instead of newlines) so you can quickly skim the algorithm and it's structure/flow just from those headings. But I agree `// Return` in itself it not a very useful comment :). I'll can replace it by a slightly more useful heading, e.g.  `// Valid result.` (to indicate the algorithm early-returned all invalids), or remove it (/replace it by a newline) if you prefer ;).